tests/test_utils.py

```diff
@@ -461,6 +462,18 @@ def test_smart_deepcopy_collection(collection, mocker):
     assert smart_deepcopy(collection) is expected_value
 
 
+@pytest.mark.parametrize('error', CAUGHT_SMART_DEEPCOPY_ERRORS)
+def test_smart_deepcopy_error(error, mocker):
+    class RaiseOnBooleanOperation(str):
+        def __bool__(self):
+            raise error('raised error')
+
+    obj = RaiseOnBooleanOperation()
+    expected_value = object()
+    mocker.patch('pydantic.utils.deepcopy', return_value=expected_value)
```

Comment:
why do we need to patch this? I don't think it's necessary.

Reply:
https://github.com/pydantic/pydantic/pull/4187/commits/624a0aa0be4a15923907501d90e516e6e5b778d4

Reply:
We were mocking out the underlying method (not under test) to assert `is`. I removed, called the underlying method, and now am using `==` as the test because the real response is a different in-memory object.